beater/config.go

```diff
@@ -48,6 +49,7 @@ type Config struct {
 	SelfInstrumentation *InstrumentationConfig `config:"instrumentation"`
 	RumConfig           *rumConfig             `config:"rum"`
 	Register            *registerConfig        `config:"register"`
+	Environment         Environment            `config:"environment"`
```

Comment:
Can we avoid the term `environment` since it's so overloaded?  How about `mode`?  Are there other envs we're considering or could `experimental` be top level instead?